### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by zarva6596**
[DEMO LINK](https://zarva6596.github.io/react_goods-selector/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -14,11 +14,67 @@ const goodsFromServer = [
...
+class App extends React.Component {
+  state = {
+    selector: false,
+    close: false,
+  };
+
+  closeButton = (text) => {
+    this.setState({ selector: false });
+    this.setState({ close: false });
+    document.querySelector('.App__list--active').className = 'App__list';
```
Comment (vkryvytskyy) [inline src/App.js:26]: don't use querySelectors in React

File: src/App.js [hunk 2]
```diff
@@ -14,11 +14,67 @@ const goodsFromServer = [
...
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selector: false,
+    close: false,
```
Comment (vkryvytskyy) [inline src/App.js:20]: you need only one field in state for completing this task

File: src/App.js [hunk 3]
```diff
@@ -14,11 +14,67 @@ const goodsFromServer = [
...
+          {' '}
+          { selector }
+          <span>
+            <button
+              type="button"
+              className={close && 'App__close'}
+              onClick={this.closeButton}
+            >
+              {close}
+            </button>
```
Comment (vkryvytskyy) [inline src/App.js:59]:
```suggestion
            {!!selector && <button
              type="button"
              className={close && 'App__close'}
              onClick={this.closeButton}
            >
              {close}
            </button>}
```
if something is selected, then button is showing up, if it's false in selector, button will not show up

File: src/App.js [hunk 4]
```diff
@@ -14,11 +14,67 @@ const goodsFromServer = [
...
+          </span>
+        </h1>
+        {goodsFromServer.map(a => (
```
Comment (vkryvytskyy) [inline src/App.js:62]:
```suggestion
        {goodsFromServer.map(good => (
```
what is `a`?

File: src/App.js [hunk 5]
```diff
@@ -14,11 +14,67 @@ const goodsFromServer = [
...
+          <p>
+            <button
+              type="button"
+              className="App__list"
```
Comment (vkryvytskyy) [inline src/App.js:66]:
```suggestion
              className={selector === good ? 'selected' : ''}
```

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]
image
please align it vertically

File: src/App.js [hunk 6]
```diff
@@ -17,58 +17,44 @@ const goodsFromServer = [
...
-              onClick={this.closeButton}
-            >
-              {close}
-            </button>
-          </span>
+          <button
+            type="button"
+            style={{
+              display: 'inline-block',
+            }}
```
Comment (vkryvytskyy) [inline src/App.js:43]: don't use inline styles

File: src/App.js [hunk 7]
```diff
@@ -17,58 +17,44 @@ const goodsFromServer = [
...
+            className={selector && 'close'}
+            onClick={this.closeButton}
+          />
         </h1>
-        {goodsFromServer.map(a => (
+        {goodsFromServer.map(good => (
           <p>
```
Comment (vkryvytskyy) [inline src/App.js:49]: where is key?

File: src/App.js [hunk 8]
```diff
@@ -17,58 +17,44 @@ const goodsFromServer = [
...
-            >
-              {close}
-            </button>
-          </span>
+          <button
+            type="button"
+            style={{
+              display: 'inline-block',
+            }}
+            className={selector && 'close'}
```
Comment (vkryvytskyy) [inline src/App.js:44]: class name should describe what it makes, `visible` is more meaningful in this case

**Review by vkryvytskyy — CHANGES_REQUESTED** [review]

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**